pubtools/pulplib/_impl/fake/client.py

```diff
@@ -87,14 +87,34 @@ def search_repository(self, criteria=None):
         # callers should not make any assumption about the order of returned
         # values. Encourage that by returning output in unpredictable order
         random.shuffle(repos)
+        return self._prepare_pages(repos)
+
+    def search_distributor(self, criteria=None):
+        criteria = criteria or Criteria.true()
+        distributors = []
+
+        filters_for_criteria(criteria, Distributor)
+
+        try:
+            for repo in self._repositories:
+                for distributor in repo.distributors:
+                    if match_object(criteria, distributor):
+                        distributors.append(distributor)
```

Comment:
When using the real client, the distributor objects are always going to include a repo_id on the returned value.  But that's a new field, and any existing code using the fake client isn't going to be setting that, so the field could be missing on the returned values here.

Instead of returning the distributor directly, could we please return a copy of it with correct repo_id ensured?  e.g.

```suggestion
                        distributors.append(attr.evolve(distributor, repo_id=repo.id))
```